File: src/goldbach.py
import math

def isPrime(n):
    if n <= 1: return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0: return False
    return True

if __name__ == "__main__":
    n = int(input())
    while(n > 0):
        x = int(input())
        reps = []
        for i in range(2, int(x / 2) + 1):
            j = x - i
            if isPrime(i) and isPrime(j):
                reps.append((i, j))
        
        print(f"{x} has {len(reps)} representation(s)")
        for i in reps:
            print(f"{i[0]}+{i[1]}")
        print()
        n -= 1

# Runtime 0.26s

File: src/goldbach.cpp
#include<bits/stdc++.h>
using namespace std;

bool isprime(int n){
    if(n <= 1) return false;
    for(int i = 2; i <= sqrt(n); i++){
        if(n % i == 0) return false;
    }
    return true;
}

int main(){
    int n; cin >> n;
    while(n--){
        int x; cin >> x;
        vector<pair<int, int>> reps;
        for(int i = 2; i <= x / 2; i++){
            int j = x - i;
            if(isprime(i) && isprime(j)){
                reps.push_back({i, j});
            } 
        }

        cout << x << " has " << reps.size() << " representation(s)" << endl;
        for(const auto& pair: reps){
            cout << pair.first << "+" << pair.second << endl;
        }
        cout << endl;
    }
    return  0;
}

// Runtime 0.08s